core/src/apps/common/cbor.py

```diff
@@ -78,9 +79,17 @@ def _cbor_encode(value: Value) -> Iterable[bytes]:
         for k, v in sorted_map:
             yield k
             yield from _cbor_encode(v)
+    elif isinstance(value, Set):
+        sorted_set = sorted(set(encode(v) for v in value.array))
+        yield _header(_CBOR_ARRAY, len(sorted_set))
+        for v in sorted_set:
+            yield v
     elif isinstance(value, Tagged):
         yield _header(_CBOR_TAG, value.tag)
-        yield from _cbor_encode(value.value)
+        if value.tag == _CBOR_SET_TAG:
+            yield from _cbor_encode(Set(value.value))
```

Comment:
I would say that the most straightforward usage of sets in cbor was to supply an actual instance of Set, when I want something encoded as a set, not an instance of Tagged that would internally be turned into a Set as it is being done now.

That way, I don't need to care about what tag should be used (if any at all). Now I have to give it explicitly, because if I don't, the result would only be encoded as a `_CBOR_ARRAY` with duplicate elements removed.

Reply:
This is a very good point, thanks.